core/services/workflows/v2/engine.go

```diff
@@ -122,8 +126,44 @@ func NewEngine(cfg *EngineConfig) (*Engine, error) {
 			NewSemaphore[[]*sdkpb.SecretResponse](cfg.LocalLimits.MaxConcurrentSecretsCallsPerWorkflow),
 			cfg.WorkflowOwner,
 			cfg.WorkflowName.String(),
-			func(shares []string) (string, error) {
-				return "", errors.New("decryption not implemented in v2 engine")
+			func(shares []string, publicKeyBytes *[32]byte) (string, error) {
+				var decryptionShares []*tdh2easy.DecryptionShare
+				ct := &tdh2easy.Ciphertext{}
+				for _, share := range shares {
```

Comment:
Rather than bailing out whenever we have an invalid share, we should just filter the invalid share out -- this is because we'll get 2F+1 shares back, but up to F of those shares could have been generated by a malicious node and could therefore be invalid.

So the logic should be:
- do your marshalling and verification
- filter out anything invalid (logging but not returning the error)
- take what's valid and try to aggregate

Reply:
Got it. This was another question, about how we cannot expect to get correct response from all N nodes.
I saw that the tdh2easy.Aggregate() method is fault tolerant too, but thanks for the above comment.